factory/dtos.go

```diff
@@ -0,0 +1,74 @@
+package factory
+
+import (
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/crypto"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/data/typeConverters"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/vm"
+)
+
+// StateComponents struct holds the state components of the Elrond protocol
+type StateComponents struct {
```

Comment:
No. Should not use this model. Place the DTOs inside each component factory. This way we won't have a large file holding everything and also we will group things up in files in a logical manner.

Reply:
I think it is better this way because you have all "core" DTOs structs for those factories in the same place. Before this, they were all one after another in `structs.go` and they gave a good overview. Also, some of these DTOs are called from other factories inside the package and this way we 'decouple' things a little bit.

Reply:
We do not decouple a thing here. Can remain as it is for the moment.